config/migrate.ts

```diff
@@ -0,0 +1,197 @@
+import * as config_DEPRECATED from './config_DEPRECATED';
+import { CLIConfiguration } from './CLIConfiguration';
+import {
+  CLIConfig,
+  CLIConfig_DEPRECATED,
+  CLIConfig_NEW,
+  Environment,
+} from '../types/Config';
+import { CmsPublishMode } from '../types/Files';
+import {
+  writeConfig,
+  createEmptyConfigFile,
+  loadConfig,
+  deleteEmptyConfigFile,
+} from './index';
+import { getConfigFilePath } from './configFile';
+import {
+  GLOBAL_CONFIG_PATH,
+  DEFAULT_CMS_PUBLISH_MODE,
+  HTTP_TIMEOUT,
+  ENV,
+  HTTP_USE_LOCALHOST,
+  DEFAULT_ACCOUNT,
+  DEFAULT_PORTAL,
+} from '../constants/config';
+import { i18n } from '../utils/lang';
+
+const i18nKey = 'config.migrate';
+
+export function getConfig(
+  useHiddenConfig?: boolean,
+  configPath?: string
+): Partial<CLIConfig> | null {
+  if (useHiddenConfig) {
+    return CLIConfiguration.config;
+  }
+  return config_DEPRECATED.loadConfig(configPath);
```

Comment:
Is this calling `loadConfig` because it may not have been loaded if the user is using a hidden config?

Reply:
Yes, exactly. Confusingly, the deprecated `loadConfig` file both loads and gets the deprecated config, so it should really be named something like `loadAndGetConfig`. But I didn't see the point in changing the name now, given that we're deprecating it.